app/shop.py

```diff
@@ -0,0 +1,39 @@
+from __future__ import annotations
+
+import datetime
+from dataclasses import dataclass
+from math import ceil
+
+
+@dataclass
+class Shop:
+    name: str
+    products: dict[str, float]
+    location: list[int]
+
+    @classmethod
+    def from_dict(cls, shop: dict) -> Shop:
+        return cls(name=shop["name"],
+                   products=shop["products"],
+                   location=shop["location"])
+
+    def is_in_stock(self, products: dict[str, int]) -> bool:
+        return all([product in self.products.keys()
+                    for product, count in products.items()])
+
+    def check(self,

```

Comment:
Check is not very good name in this case. I'd suggest to rename the method to something like `print_receipt`, and the variable to `receipt`.